src/core/core.controller.js

```diff
@@ -933,7 +937,62 @@ module.exports = function(Chart) {
 			me.lastActive = me.active;
 
 			return changed;
-		}
+		},
+		/**
+		 * Format the data list if the data is missing some points in the datasets.
+		 * @private
+		 * @param {array} dataArray array to format
+		 * @return {array} the formated array
+		 */
+		formatDataset: function(dataArray) {
+			var labels = this.chart.data.labels;
+			var tmp = dataArray.slice(0);
+			var result = dataArray;
+			var match = this.dataInLabel(tmp, labels);
+			if (match && dataArray.length < labels.length) {
+				for (var i = 0; i < labels.length; i++) {
+					var label = labels[i];
+					result[i] = {x: label, y: this.getY(label, tmp)};
+				}
+			}
+			return result;
+		},
+		/**
+		 * Return the y data of the label, if no this label, return null
+		 * @private
+		 * @param {string} label the label in the labels array
+		 * @param {array} dataArray the data in the datasets.
+		 * @return {number} the y data according to the label
+		 */
+		getY: function(label, dataArray) {
+			var y = null;
+			for (var i = 0; i < dataArray.length; i++) {
+				var item = dataArray[i];
+				if (item.x === label) {
+					y = item.y;
+				}
+			}
+			return y;
+		},
+		/**
+		 * Find if the data in datasets exists in the labels
+		 * @private
+		 * @param {array} dataArray the data in the datasets
+		 * @param {array} labelArray the label in the labels array
+		 * @return {boolean} the match result, true is included
+		 */
+		dataInLabel: function(dataArray, labelArray) {
```

Comment:
this can be put with the other functions (line 25-68) since it doesn't use any class members